gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireStore.java

```diff
@@ -2139,6 +2139,9 @@ private void clearStatics(boolean forBoot) {
       // Instead do this at the boot time itself
       CallbackProcedures.clearStatics();
       DistributionObserver.clearStatics();
+      if (!forBoot) {
+        FabricDatabase.TEST_INDEX_RECREATE = false;
+      }
```

Comment:
This flag handling in product is awkwark. Can the tests instead do something like changing the size in irf explicitly (perhaps create index twice with different sizes, copy older irf somewhere, then overwrite it over new irf after stop of the VM).